service-base/src/main/java/org/eclipse/hono/service/auth/HonoSaslAuthenticatorFactory.java

```diff
@@ -40,11 +40,16 @@
      * 
      * @param vertx the Vertx environment to run the factory in.
      * @param validator The object to use for validating auth tokens.
+     * @param actualAuthenticationService The authentication service that will handle the authentication requests sent
+     *            on the Vert.x event bus. Used for determining the supported SASL mechanisms.
      * @throws NullPointerException if any of the parameters is {@code null}.
      */
     @Autowired
-    public HonoSaslAuthenticatorFactory(final Vertx vertx, @Qualifier(AuthenticationConstants.QUALIFIER_AUTHENTICATION) final AuthTokenHelper validator) {
-        this(new EventBusAuthenticationService(vertx, validator));
+    public HonoSaslAuthenticatorFactory(final Vertx vertx,
```

Comment:
this is public API so we cannot simply change it in a non backward compatible way. Apart from that, IMHO the restriction of the mechanisms is not so much a matter of the capabilities of the underlying `AuthenticationService` but instead is a matter of configuration. The `FileBasedAuthenticationService` can very well authenticate a client based on the certificate's subject DN. However, in certain scenarios you do not want the `SaslAuthenticator` sitting in front of it to offer the *EXTERNAL* mechanism because it doesn't make sense in the given system setup. IMHO we should therefore simply add constructors to this class which also accept a list of supported mechanisms and which is then used to configure the `SaslAuthenticator`.

Reply:
I agree that it would be better to have the `supportedSaslMechanisms` be configurable via a config property. I'll have a look at that and update this PR.

Reply:
> IMHO we should therefore simply add constructors to this class which also accept a list of supported mechanisms and which is then used to configure the `SaslAuthenticator`.

I would rather have the supported mechanisms be exposed via an `AuthenticationService#getSupportedSaslMechanisms()` method, because I see the corresponding config property as belonging to the config properties of the authentication service as well. See the 2nd commit I just added for the added config properties. (I first wanted to put these in a common base class but this doesn't work because of the existing inheritance hierarchy in `AuthenticationServerClientConfigProperties`).

I've amended the first commit to add a constructor to `HonoSaslAuthenticatorFactory` matching the current one to keep backwards compatibility. I've also added some checks, skipping stuff if one or the other mechanism isn't configured to be supported.